From f206e9120be5942d0ba49b19b94c0d30d8f0e79b Mon Sep 17 00:00:00 2001
From: Hardik T Shah <hardik.t.shah@intel.com>
Date: Thu, 10 Mar 2016 22:30:12 +0530
Subject: [PATCH 3866/4301] SoundWire: Intel: Implement method to set the port
 params

commit 7223b6638b496cf140a2a44fca0b85a7c2853b89 from
git://git@github.com:01org/linux-intel-4.9.git

This method implements method to set the SoundWire port
params in device specific register map.

Change-Id: I3be29f3fc05ac4f1c2d924201c02db795aca1807
Signed-off-by: Hardik T Shah <hardik.t.shah@intel.com>
---
 drivers/sdw/sdw_cnl.c |   23 +++++++++++++++++++++++
 1 files changed, 23 insertions(+), 0 deletions(-)

diff --git a/drivers/sdw/sdw_cnl.c b/drivers/sdw/sdw_cnl.c
index 7405c87..f07d33d 100644
--- a/drivers/sdw/sdw_cnl.c
+++ b/drivers/sdw/sdw_cnl.c
@@ -748,6 +748,29 @@ static int cnl_sdw_set_clock_freq(struct sdw_master *mstr,
 static int cnl_sdw_set_port_params(struct sdw_master *mstr,
 			struct sdw_port_params *params, int bank)
 {
+	struct cnl_sdw *sdw = sdw_master_get_drvdata(mstr);
+	struct cnl_sdw_data *data = &sdw->data;
+	int dpn_config = 0, dpn_config_offset;
+
+	if (bank)
+		dpn_config_offset = SDW_CNL_DPN_CONFIG1;
+	else
+		dpn_config_offset = SDW_CNL_DPN_CONFIG0;
+
+	dpn_config = cnl_sdw_port_reg_readl(data->sdw_regs,
+				dpn_config_offset, params->num);
+
+	dpn_config |= (((params->word_length - 1) & DPN_CONFIG_WL_MASK) <<
+				DPN_CONFIG_WL_SHIFT);
+	dpn_config |= ((params->port_flow_mode & DPN_CONFIG_PF_MODE_MASK) <<
+				DPN_CONFIG_PF_MODE_SHIFT);
+	dpn_config |= ((params->port_data_mode & DPN_CONFIG_PD_MODE_MASK) <<
+				DPN_CONFIG_PD_MODE_SHIFT);
+	cnl_sdw_port_reg_writel(data->sdw_regs,
+				dpn_config_offset, params->num, dpn_config);
+
+	cnl_sdw_port_reg_readl(data->sdw_regs,
+				dpn_config_offset, params->num);
 	return 0;
 }
 
-- 
1.7.5.4

